### DISCUSSION TITLE
Matic chainId

### DISCUSSION THREAD

**Discussion by benjaminpayoux** [Web3Auth/Web3Auth#972] [Help]
Hi,

When I initialize Torus like this:

```
this.torus = new Torus()
        await this.torus.init({
            network: {
                host: 'matic'
            },
            showTorusButton: true,
            enableLogging: true
        })
        await this.torus.login()
```

If I log `this.torus.provider`, the chainId is 1 instead of 137.

Only a refresh correctly setup the chainId

Thank you for your help

**Comment by atchrispinto** [comment]
Which docs are you referring to?

**Reply by benjaminpayoux** [reply]
This one : https://docs.tor.us/wallet/api-reference/installation

**Reply by atchrispinto** [reply]
Try this https://docs.tor.us/wallet/api-reference/ethereum-api#setprovider 

In the first snippet you sent, you use await, not a case where you forget an async keyword right?

**Reply by benjaminpayoux** [reply]
I double checked and I do not think I missed an await. Weird thing is that before switching network, if I refresh, I am on the correct one.
Set provider method triggers a popup, what we want to avoid.
